tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php

```diff
@@ -0,0 +1,282 @@
+			->willReturn( [
+			    'response' => [
+			        'code' => 200,
+			    ],
+			    'body' => 'null',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test' );
+
+		// Assert
+		$this->assertEquals( 1, $result );
+	}
+
+	public function test_http_request__wrong_server_response() {
+		// Arrange
+		$this->http_stub->method( 'request' )
+			->willReturn( [
+			    'response' => [
+			        'code' => 200,
+			    ],
+			    'body' => 'false',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test' );
+
+		// Assert
+		$this->assertInstanceOf( \WP_Error::class, $result );
+	}
+
+	public function test_http_request__error_from_server() {
+		// Arrange
+		$this->http_stub->method( 'request' )
+			->willReturn( [
+			    'response' => [
+			        'code' => 422,
+			    ],
+			    'body' => '{"message": "my test error"}',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test' );
+
+		// Assert
+		$this->assertInstanceOf( \WP_Error::class, $result );
+		$this->assertEqualSets( [
+			'422' => ['my test error']
+		], $result->errors );
+	}
+
+	public function test_http_request__as_array() {
+		// Arrange
+		$this->http_stub->method( 'request' )
+			->willReturn( [
+			    'response' => [
+			        'code' => 200,
+			    ],
+			    'body' => '{"message": "as array"}',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test', [], [ 'return_type' => Base_App::HTTP_RETURN_TYPE_ARRAY ] );
+
+		// Assert
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEqualSets( [ 'message' => 'as array' ], $result );
+	}
+
+	public function test_http_request__with_connect() {
+		// Arrange
+		$endpoint = 'test/1';
+		$user = $this->get_connected_user();
+
+		$this->http_stub->method( 'request' )
+			->with(
+			    $this->equalTo( Mock_App::BASE_URL . '/' . $endpoint ),
+			    $this->equalTo( [
+			        'headers' => [
+			        	'access-token' => 'access_token_test',
+				        'app' => 'mock-app',
+				        'client-id' => 'client_id_test',
+				        'endpoint' => $endpoint,
+				        'home-url' => 'http://example.org/',
+				        'local-id' => $user->ID,
+						'site-key' => 'site_key_test',
+				        'X-Elementor-Signature' => '2f5dccc6fd53fa38f80d18fbdd9e02a5c0e1dde77fe012e5e11679a9ff0a6faa',
+				        'a' => '1',
+			        ],
+			        'method' => 'PUT',
+			        'timeout' => 30,
+			        'body' => [
+			            'b' => '2',
+			        ],
+			    ] )
+			)
+			->willReturn( [
+			    'response' => [
+			        'code' => 200,
+			    ],
+			    'body' => '{"status": "success"}',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'PUT', $endpoint, [
+			'timeout' => 30,
+			'headers' => [
+				'a' => '1',
+			],
+			'body' => [
+				'b' => '2',
+			]
+		] );
+
+		// Assert
+		$this->assertEquals( (object) [ 'status' => 'success' ], $result );
```

Comment:
I didn't understand the test (why or how will the connect headers affect it?)

Reply:
The `with` method of the mock checks that the right parameters are sent to the API. 
I will add some comments

Reply:
Done